.github/workflows/npmpublish.yml

```diff
@@ -3,14 +3,16 @@ on:
   push:
     branches:
       - main
+  pull_request:
+    branches: [main]
 
 jobs:
   publish:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3.1.0
       - uses: ./.github/actions/pnpm-setup-action
-      - run: pnpm -r publish --access public
+      - run: pnpm -r publish --access public ${{ (github.event_name == 'pull_request' && '--dry-run') || '' }}
```

Comment:
[Currently fails](https://github.com/Sage-Bionetworks/synapse-web-monorepo/actions/runs/5394301654/jobs/9795278389) with 

> ERR_PNPM_GIT_UNKNOWN_BRANCH  The Git HEAD may not attached to any branch, but your "publish-branch" is set to "master|main"

Do we care for the dry-run?  If not, maybe add `--no-git-checks` to this command so that it doesn't check if current branch is the publish branch